homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -65,53 +76,97 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 
     envoy_reader = EnvoyReader(ip_address, username, password)
 
+    async def async_update_data():
+        """Fetch data from API endpoint."""
+        data = {}
+        async with async_timeout.timeout(30):
+            try:
+                await envoy_reader.getData()
+            except httpx.HTTPError as err:
+                raise UpdateFailed(f"Error communicating with API: {err}") from err
+
+            for condition in monitored_conditions:
+                if condition != "inverters":
+                    data[condition] = await getattr(envoy_reader, condition)()
```

Comment:
Does this call not do I/O?

Reply:
This call should not do I/O. The `getData()` method should be the only method that causes I/O.

But looking into the code `envoy_reader` code there is still a path that does a HTTP request.  I'll fix it in the library

Reply:
Bumped the API version to 0.18.1. Fixed the I/O issue, so only getData() and no other methods called from the Home Assistant sensor cause I/O